### PR TITLE
Conditionally disable SVE extension support in LLDB's NativeRegisterC…

### PR DESCRIPTION
**Pull request by adrian-prantl**
…ontext.

On Amazon Linux 2, the ptrace header doesn't define the data
structures for SVE support. This patch detects this and disables SVE
support if necessary.

Upstream LLDB doesn't depend on this definition any more. This is just
a stopgap for the stable-20210107 branch.

rdar://78983055

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp [hunk 1]
```diff
@@ -1095,13 +1114,16 @@ void NativeRegisterContextLinux_arm64::ConfigureRegisterContext() {
...
+#endif
 }
 
 uint32_t NativeRegisterContextLinux_arm64::CalculateFprOffset(
     const RegisterInfo *reg_info) const {
   return reg_info->byte_offset - GetGPRSize();
 }
 
+#if LLDB_HAVE_USER_SVE_HEADER
+xxx
```
Comment (drexin) [inline lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp]: Should this be here?

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux_arm64.cpp]: Thanks for catching this! This is a leftover from me trying that the conditional actually works.

**Review by drexin — APPROVED** [review]
Thank you!

**Pull request merged**